src/dawn/IIR/IIRSerializer.cpp

```diff
@@ -0,0 +1,799 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+#include "dawn/IIR/IIRSerializer.h"
+#include "dawn/IIR/IIR.pb.h"
+#include "dawn/IIR/IIRNodeIterator.h"
+#include "dawn/IIR/MultiStage.h"
+#include "dawn/IIR/StatementAccessesPair.h"
+#include "dawn/IIR/StencilInstantiation.h"
+#include "dawn/Optimizer/PassComputeStageExtents.h"
+#include "dawn/Optimizer/PassInlining.h"
+#include "dawn/SIR/ASTVisitor.h"
+#include "dawn/SIR/SIR.h"
+#include "dawn/Support/ASTSerializer.h"
+#include <fstream>
+#include <google/protobuf/util/json_util.h>
+#include <list>
+#include <stack>
+#include <tuple>
+#include <utility>
+
+namespace dawn {
+static void setAccesses(proto::iir::Accesses* protoAccesses,
+                        const std::shared_ptr<iir::Accesses>& accesses) {
+  auto protoReadAccesses = protoAccesses->mutable_readaccess();
+  for(auto IDExtentsPair : accesses->getReadAccesses()) {
+    proto::iir::Extents protoExtents;
+    for(auto extent : IDExtentsPair.second.getExtents()) {
```

Comment:
somehow for accesses with extent 0, the proto message of Extents is not being filled, I dont know why, but we need to look into it